pkg/operator/operator.go

```diff
@@ -143,6 +158,50 @@ func NewOperator(ctx context.Context, operator *operator.Operator) (context.Cont
 	}
 }
 
+// Copied from karpenter-core pkg/operator/operator.go, needed for webhooks
+func (o *Operator) WithControllers(ctx context.Context, controllers ...controller.Controller) *Operator {
+	for _, c := range controllers {
+		lo.Must0(c.Register(ctx, o.Manager))
+	}
+	return o
+}
+
+// Copied from karpenter-core pkg/operator/operator.go, needed for webhooks
+func (o *Operator) WithWebhooks(ctx context.Context, ctors ...knativeinjection.ControllerConstructor) *Operator {
+	if !karpenteroptions.FromContext(ctx).DisableWebhook {
+		o.webhooks = append(o.webhooks, ctors...)
+		lo.Must0(o.Manager.AddReadyzCheck("webhooks", webhooksalt.HealthProbe(ctx)))
+		lo.Must0(o.Manager.AddHealthzCheck("webhooks", webhooksalt.HealthProbe(ctx)))
+	}
+	return o
+}
+
+// Copied from karpenter-core pkg/operator/operator.go, needed for webhooks
+func (o *Operator) Start(ctx context.Context, cp cloudprovider.CloudProvider) {
+	wg := &sync.WaitGroup{}
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		lo.Must0(o.Manager.Start(ctx))
+	}()
+	if karpenteroptions.FromContext(ctx).DisableWebhook {
+		log.FromContext(ctx).Info("conversion webhooks are disabled")
+	} else {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			// Taking the first supported NodeClass to be the default NodeClass
+			gvk := lo.Map(cp.GetSupportedNodeClasses(), func(nc status.Object, _ int) schema.GroupVersionKind {
+				return object.GVK(nc)
+			})
+			ctx = injection.WithNodeClasses(ctx, gvk)
+			ctx = injection.WithClient(ctx, o.GetClient())
+			webhooksalt.Start(ctx, o.GetConfig(), o.webhooks...) // This is our alt copy of webhooks that can support multiple apiservers
```

Comment:
This is now in self-hosted codepath as well ... Any issues expected, or should it just work? Edit: Confirmed, works